Realm/RLMRealmConfiguration.mm

```diff
@@ -25,6 +25,9 @@
 
 #import "schema.hpp"
 #import "shared_realm.hpp"
+#import "sync/sync_config.hpp"
+
+#include <realm/util/optional.hpp>
```

Comment:
This doesn't appear to be necessary. `SyncConfig` has a `util::Optional` as a member so is required to have included this header.